.github/workflows/unittest-existence.yml

```diff
@@ -35,43 +35,12 @@ permissions:
   contents: read
   pull-requests: write
 
-env:
-    comment: "Thank you for your pull request! We could not find a unittest file for this change. Please consider adding some. If it is hard to unittest, please add the 'Hard to Unit Test' label. If unit tests are not needed, please add the 'Unit Test Not Needed' label."
-
-
 jobs:
   unittest:
     name: Check if Unit Test being added.
     if: ${{ !contains(github.event.pull_request.labels.*.name, 'Unit Test Not Needed') && !contains(github.event.pull_request.labels.*.name, 'Hard to Unit Test')}}
-    runs-on: ubuntu-latest
-    steps:
-      - name: Check if unittest file was added
-        uses: dorny/paths-filter@v3
-        id: unittest_existence
-        with:
-          token: ${{ secrets.GITHUB_TOKEN }}
-          filters: |
-            exists:
-              - added|modified: 'tests/unit/**'
-
-      - name: Check for existing comment
-        if: steps.unittest_existence.outputs.exists == 'false'
-        uses: peter-evans/find-comment@v3
-        id: find_comment
-        with:
-          issue-number: ${{ github.event.pull_request.number }}
-          comment-author: "github-actions[bot]"
-          body-includes: ${{ env.comment }}
-      - name: Comment if no unittest file was added
-        if: steps.unittest_existence.outputs.exists == 'false' &&
-          steps.find_comment.outputs.exists == 'false'
-        run: |
-          gh issue comment ${{ github.event.pull_request.number }} --repo ${{ github.repository }} --body "${{ env.comment }}"
-        env:
-          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
-      - name: Fail job if unittest is missing and required
-        if: steps.unittest_existence.outputs.exists == 'false'
-        uses: actions/github-script@v7
-        with:
-          script: core.setFailed('Unit Test required to merge.')
+    uses: dbt-labs/actions/.github/workflows/check-change-and-comment.yml@main
```

Comment:
With the new action things should look much easier.
@emmyoop thoughts on this refactor and how to best test it?